Framework/Algorithms/src/FitPeaks.cpp

```diff
@@ -0,0 +1,2301 @@
+      throw std::runtime_error("Peak tolerance out of index");
+    postol = m_peakPosTolerances[peakindex];
+  }
+
+  // get peak position and analyze the fitting is good or not by various
+  // criteria
+  double peak_pos = fitfunction.peakfunction->centre();
+  bool good_fit(false);
+  if ((cost < 0) || (cost >= DBL_MAX - 1.)) {
+    // unphysical cost function value
+    peak_pos = -4;
+  } else if (fitfunction.peakfunction->height() < m_minPeakHeight) {
+    // peak height is under minimum request
+    peak_pos = -3;
+  } else if (case23) {
+    // case b and c to check peak position without defined peak tolerance
+    std::pair<double, double> fitwindow = getPeakFitWindow(wsindex, peakindex);
+    if (fitwindow.first < fitwindow.second) {
+      // peak fit window is specified or calculated: use peak window as position
+      // tolerance
+      if (peak_pos < fitwindow.first || peak_pos > fitwindow.second) {
+        // peak is out of fit window
+        peak_pos = -2;
+        g_log.debug() << "Peak position " << peak_pos << " is out of fit "
+                      << "window boundary " << fitwindow.first << ", "
+                      << fitwindow.second << "\n";
+      } else
+        good_fit = true;
+    } else {
+      // use the 1/2 distance to neiboring peak without defined peak window
+      double left_bound(-1);
+      if (peakindex > 0)
+        left_bound = 0.5 * (expected_peak_positions[peakindex] -
+                            expected_peak_positions[peakindex - 1]);
+      double right_bound(-1);
+      if (peakindex < m_numPeaksToFit - 1)
+        right_bound = 0.5 * (expected_peak_positions[peakindex + 1] -
+                             expected_peak_positions[peakindex]);
+      if (left_bound < 0)
+        left_bound = right_bound;
+      if (right_bound < 0)
+        right_bound = left_bound;
+      if (left_bound < 0 || right_bound < 0)
+        throw std::runtime_error("Code logic error such that left or right "
+                                 "boundary of peak position is negative.");
+      if (peak_pos < left_bound || peak_pos > right_bound)
+        peak_pos = -2.5;
+      else
+        good_fit = true;
+    }
+  } else if (fabs(fitfunction.peakfunction->centre() -
+                  expected_peak_positions[peakindex]) > postol) {
+    // peak center is not within tolerance
+    peak_pos = -5;
+    g_log.debug() << "Peak position difference "
+                  << fabs(fitfunction.peakfunction->centre() -
+                          expected_peak_positions[peakindex])
+                  << " is out of range of tolerance: " << postol << "\n";
+  } else {
+    // all criteria are passed
+    good_fit = true;
+  }
+
+  // set cost function to DBL_MAX if fitting is bad
+  double adjust_cost(cost);
+  if (!good_fit) {
+    // set the cost function value to DBL_MAX
+    adjust_cost = DBL_MAX;
+  }
+
+  // double peak_position = peak_pos;
+
+  // reset cost
+  if (adjust_cost > DBL_MAX - 1) {
```

Comment:
Why isn't this folded into the check on line 1050? Do you ever get a "good fit" that has an `adjust_cost > DBL_MAX`?

Reply:
I am afraid of some case that a "good fit" with error as DBL_MAX.  I don't want to have this case (thought not likely to happen) slip through.